tests/sentry/workflow_engine/endpoints/test_project_detector_index.py

```diff
@@ -55,22 +59,30 @@ def setUp(self):
         self.valid_data = {
             "name": "Test Detector",
             "detectorType": MetricAlertFire.slug,
-            "dataSource": {
-                "queryType": SnubaQuery.Type.ERROR.value,
-                "dataset": Dataset.Events.name.lower(),
-                "query": "test query",
-                "aggregate": "count()",
-                "timeWindow": 60,
-                "environment": self.environment.name,
-                "eventTypes": [SnubaQueryEventType.EventType.ERROR.name.lower()],
-            },
-            "dataConditions": [
+            "dataSources": [
                 {
-                    "type": Condition.GREATER,
-                    "comparison": 100,
-                    "result": DetectorPriorityLevel.HIGH,
-                }
+                    "queryType": SnubaQuery.Type.ERROR.value,
+                    "dataset": Dataset.Events.name.lower(),
+                    "query": "test query",
+                    "aggregate": "count()",
+                    "timeWindow": 60,
+                    "environment": self.environment.name,
+                    "eventTypes": [SnubaQueryEventType.EventType.ERROR.name.lower()],
+                },
             ],
+            "conditionGroup": {
+                "id": self.data_condition_group.id,
+                "organizationId": self.organization.id,
+                "logicType": self.data_condition_group.logic_type,
```

Comment:
similarly, did i misunderstand having a configurable logic type here? don't think the ui allows configuring this at the moment

Reply:
The existing endpoints we have pass along the entire payload of existing data with the changes so having it here doesn't necessarily mean it needs to be configurable.

However, one question I had about this is that `workflow_condition_group` on the `Detector` model (a `DataConditionGroup`) is nullable, so is there a case where we have a detector w/o one and we'd be adding it at this stage?